syntax = "proto3";

package sendfile;

message Request{
    bytes resource = 2;
}

message Response{
    int32 status_code = 1;
    string response = 2;
}

message RequestSimple{
    string resource = 3;
    string fileName = 7;
}

message fileRequest{
    string urlCopy = 1;
    string fileName = 2;
    string partitionName = 3;
    bytes content = 4;
}

message fileResponse{
    int32 status_code = 4;
}

message copyRequest{
    string fileName = 1;
    string partitionName = 2;
    bytes content = 3;
}

message copyResponse{
    int32 status_code = 4;
}

message ResponseSimple{
    int32 status_code = 4;
    bytes response = 3;
}

message openRequest{
    string fileName = 5;
    string mode = 6;
}

message openResponse{
    int32 status_code = 9;
    string response = 5;
}

message readRequest{
    string fileName = 7;
    string partName = 12;
}

message readResponse{
    int32 status_code = 10;
    bytes response = 8;
}

message writeRequest{
    string fileName = 11;
    string partName = 12;
    bytes data = 13;
}

message writeResponse{
    int32 status_code = 14;
}

service ProductService{
    rpc clientStreaming(stream Request) returns (Response) {}
    rpc clientSingle(RequestSimple) returns (ResponseSimple) {}
    rpc open(openRequest) returns (openResponse) {}
    rpc read(readRequest) returns (readResponse) {}
    rpc write(writeRequest) returns (writeResponse) {}
    rpc sendFile(fileRequest) returns (fileResponse) {}
    rpc copyPart(copyRequest) returns (copyResponse) {}
}